app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,8 +86,23 @@ EOF
 elif [ -f /etc/debian_version ]; then
   <%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
 <% if @repo_gpg_key_url.present? -%>
+<%
+  <% # Ubuntu 16.04 does not support GPG public keys in "/etc/apt/trusted.gpg.d/" -%>
+  use_apt_key =
+    if @host.operatingsystem.name == 'Ubuntu' && @host.operatingsystem.major.split('.')[0].to_i == 16
+      true
+    else
+      false
+    end
+%>
+
   apt-get -y install ca-certificates gpg
```

Comment:
You install `gpg` only for `apt-key`, so that part should have been inside the `if` condition. `ca-certificates` is for curl and should remain outside of it.